AdvancedStatistics: disregard suspended, etc. cards

draeklae opened the pull request:
## Fixes
Fixes #5289 
Suspended cards are ignored by prediction algorithm.

mikehardy commented:
Seems to match the wisdom on which queue numbers mean what here - all the buried/suspended ones are negative - https://github.com/ankidroid/Anki-Android/wiki/Database-Structure#database-schema

@draeklae the commit is small so I can accept skipping much of the template, but how did you test this?

draeklae commented:
Regarding tests: Started a deck from scratch, added 4~5 cards and did some reviews. Enabled advanced statistics in the options, then compared the prediction graphs before and after suspending a couple cards.

In retrospect, buried cards should *not* be simply ignored since there may be new sibling cards or, also, user-buried cards—which *are* relevant for prediction. However, handling that needs considerably more changes than this one-line commit (and is probably better handled in a separate PR). 

Updated PR so as to address only the suspended card bug.

mikehardy commented:
@draeklae thanks for your patience on this as I was out traveling and I've been super focused on fixing crashes, so this sat idle. This looks like a really focused fix now and I appreciate the testing - I'll merge it and it'll be available in the next day or three as alpha48 🏆

mikehardy commented:
2.10alpha48 is published to google and will be available with this as soon as google is done processing 🚀

mikehardy commented:
Hello! with the 2.14 release we started accepting donations with the stated goal of accepting money from users (the users' way of saying "Thanks!") to support maintenance and development of AnkiDroid.

We are looking through PRs merged in releases going back to release 2.10 (where we have good record keeping of what's in releases) to let our contributors know.

If you'd like to submit an expense related to your work here, to share in a portion of our hopefully-sustainable monthly disbursement, please read here and do so in the next couple of days:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process

This is our first time handling donations and attempting to compensate anyone with these processes, so please be flexible and patient with us

Thank you for your contribution to AnkiDroid!

mikehardy commented:
Hello! with the 2.14 release we started accepting donations with the stated goal of accepting money from users (the users' way of saying "Thanks!") to support maintenance and development of AnkiDroid.

We are looking through PRs merged in releases going back to release 2.10 (where we have good record keeping of what's in releases) to let our contributors know.

If you'd like to submit an expense related to your work here, to share in a portion of our hopefully-sustainable monthly disbursement, please read here and do so in the next couple of days:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process

This is our first time handling donations and attempting to compensate anyone with these processes, so please be flexible and patient with us

Thank you for your contribution to AnkiDroid!
